VideoLocker/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -348,4 +397,336 @@ public boolean onCreateOptionsMenu(Menu menu) {
+                                context.getString(R.string.error_account_not_linked_desc_fb)));
+                    }
+                } else if (backend.equalsIgnoreCase(PrefManager.Value.BACKEND_GOOGLE)) {
+                    social = api.registerByFaceBook(accessToken);
+
+                    if ( social.getStatus() == RegisterResponse.Status.ERROR ) {
+                        throw new LoginException(new LoginErrorMessage(
+                                getString(R.string.error_account_not_linked_title_google),
+                                getString(R.string.error_account_not_linked_desc_google)));
+                    }
+                }
+                    // we got a valid accessToken so profile can be fetched
+                    ProfileModel profile =  api.getProfile();
+
+                    // store profile json
+                    if (profile != null ) {
+                        PrefManager pref = new PrefManager(context, PrefManager.Pref.LOGIN);
+                        pref.put(PrefManager.Key.PROFILE_JSON,  profile.json);
+                        pref.put(PrefManager.Key.AUTH_TOKEN_BACKEND, null);
+                        pref.put(PrefManager.Key.AUTH_TOKEN_SOCIAL, null);
+                    }
+
+                    if (profile.email != null) {
+                        // we got valid profile information
+                        return profile;
+                    }
+
+                throw new LoginException(new LoginErrorMessage(
+                        getString(R.string.login_error),
+                        getString(R.string.login_failed)));
+            } catch (Exception e) {
+                logger.error(e);
+                handle(e);
+            }
+            return null;
+        }
+
+    }
+
+    android.view.View.OnClickListener facebookClickListener = new View.OnClickListener() {
+
+        @Override
+        public void onClick(View v) {
+            if (AppConstants.offline_flag) {
+                showErrorMessage(getString(R.string.no_connectivity),
+                        getString(R.string.network_not_connected));
+            } else {
+                Task<Void> logout = new Task<Void>(RegisterActivity.this) {
+
+                    @Override
+                    protected Void doInBackground(Object... arg0) {
+                        try {
+                            socialLoginDelegate.socialLogout(SocialFactory.TYPE_FACEBOOK);
+                        } catch(Exception ex) {
+                            // no need to handle this error
+                            logger.error(ex);
+                        }
+                        return null;
+                    }
+
+                    @Override
+                    public void onFinish(Void result) {
+                        socialLoginDelegate.socialLogin(SocialFactory.TYPE_FACEBOOK);
+                    }
+
+                    @Override
+                    public void onException(Exception ex) {
+
```

Comment:
Should we at least log exceptions? Or can we not have any exceptions because we wrap ``doInBackground``? Maybe we should get rid of that ``try`` ``catch`` and just log here, since we have to implement it anyway?